+jrclust/+detect/@DetectController/detectOneRecording.m

```diff
@@ -171,64 +130,111 @@
             if ~isempty(impSites)
                 iSpikeSites = impSites(inInterval);
             end
-        end
+        end   
+        
+        if ~isempty(impTimes) && ~any(inInterval) 
```

Comment:
You can invert this condition and put everything in the `else` block underneath it, say

```matlab
if isempty impTimes || any(inInterval)
    % load raw samples
    iSamplesRaw = hRec.readRawROI(obj.hCfg.siteMap, 1+loadOffset:loadOffset+nSamples);
    ...
```
```

Reply:
Done.